### PR TITLE
Update JQuery UI Layout

### PR DESCRIPTION
**Pull request by drarmstr**
Update JQuery UI Layout typings to provide proper types instead of "any" for most options.

This is an updated pull request from #7372 properly based on a git branch.

### CODE REVIEW

File: jquery.ui.layout/jquery.ui.layout.d.ts [hunk 1]
```diff
@@ -2,39 +2,484 @@
 // Project: http://layout.jquery-dev.net/
 // Definitions by: Steve Fenton <https://github.com/Steve-Fenton>
 // Definitions: https://github.com/borisyankov/DefinitelyTyped
+// Definitions updated by: Douglas Armstrong <https://github.com/drarmstr>
```
Comment (vvakame) [inline jquery.ui.layout/jquery.ui.layout.d.ts]: please use this format. https://github.com/DefinitelyTyped/definition-header can extract your name.
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/jquery/jquery.d.ts#L3

Comment (drarmstr) [inline jquery.ui.layout/jquery.ui.layout.d.ts]: Ok, I'll update it as requested.

File: jquery.ui.layout/jquery.ui.layout.d.ts [hunk 2]
```diff
@@ -1,40 +1,484 @@
 // Type definitions for jQuery UI Layout Plug-in
 // Project: http://layout.jquery-dev.net/
-// Definitions by: Steve Fenton <https://github.com/Steve-Fenton>
+// Definitions by: Douglas Armstrong <https://github.com/drarmstr>
```
Comment (vvakame) [inline jquery.ui.layout/jquery.ui.layout.d.ts]: sorry, one more things. please 'append' your name below original author.
like this https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/jquery/jquery.d.ts#L3

Comment (drarmstr) [inline jquery.ui.layout/jquery.ui.layout.d.ts]: Ok, good; I like that better.  Header is updated.

**Pull request merged**